hytra/core/hypothesesgraph.py

```diff
@@ -173,16 +173,21 @@ def checkNodeWhileAddingLinks(frame, obj):
             if (frame, obj) not in self._graph:
                 getLogger().warning("Adding node ({}, {}) when setting up links".format(frame, obj))
 
-        kdTreeNextFrame = None
+        kdTreeFrames = [None]*(skipLinks+1)
+        # kdTreeNextFrame = None
         for frame in range(len(probabilityGenerator.TraxelsPerFrame.keys()) - 1):
             if frame > 0:
-                kdTreeThisFrame = kdTreeNextFrame
+                del kdTreeFrames[0] # this is the current frame
+                if frame + skipLinks < len(probabilityGenerator.TraxelsPerFrame.keys()):
+                    kdTreeFrames.append(self._buildFrameKdTree(probabilityGenerator.TraxelsPerFrame[frame + skipLinks]))
+                    self._addNodesForFrame(frame + skipLinks, probabilityGenerator.TraxelsPerFrame[frame + skipLinks])
+                else:
+                    kdTreeFrames.append(self._buildFrameKdTree(probabilityGenerator.TraxelsPerFrame[2*frame - skipLinks - len(probabilityGenerator.TraxelsPerFrame.keys()) + 1]))
```

Comment:
I think you are right :+1:  Until I do not find my reason again, I will delete those lines. It would have made sens for the Links, not for the nodes...